video_xblock/static/js/video_xblock.js

```diff
@@ -46,12 +47,23 @@ function VideoXBlockStudentViewInit(runtime, element) {
       // Discard a message received from another domain
       return;
     try {
+      if (event.data.action === "saveState") {
+        updateTranscriptDownloadUrl(event.data.downloadTranscriptUrl);
```

Comment:
You **MUST NOT** not rely on saveState's to update transcript download url. It's an independent plugin that can always be disabled.